jenkinsfile-mode.el

```diff
@@ -236,10 +236,66 @@ Run this manually when editing this file to get an updated the list of keywords.
 (defvar jenkinsfile-mode-font-lock-defaults
   (append groovy-font-lock-keywords jenkinsfile-mode-font-lock-keywords))
 
+(defun jenkinsfile-pipeline-step-compeletion-at-point ()
+  "completion for pipeline step"
+  (interactive)
+  (let* (
+         (bds (bounds-of-thing-at-point 'symbol))
+         (start (car bds))
+         (end (cdr bds)))
+    (list start end jenkinsfile-mode--pipeline-step-keywords . nil ))
+  )
+
+(defun jenkinsfile-core-step-compeletion-at-point ()
+  "completion for core step"
+  (interactive)
+  (let* (
+         (bds (bounds-of-thing-at-point 'symbol))
+         (start (car bds))
+         (end (cdr bds)))
+    (list start end jenkinsfile-mode--core-step-keywords . nil ))
+  )
+
+(defun jenkinsfile-option-compeletion-at-point ()
+  "completion for option step"
+  (interactive)
+  (let* (
+         (bds (bounds-of-thing-at-point 'symbol))
+         (start (car bds))
+         (end (cdr bds)))
+    (list start end jenkinsfile-mode--option-keywords . nil ))
+  )
+
+(defun jenkinsfile-directive-compeletion-at-point ()
+  "completion for directive step"
+  (interactive)
+  (let* (
+         (bds (bounds-of-thing-at-point 'symbol))
+         (start (car bds))
+         (end (cdr bds)))
+    (list start end jenkinsfile-mode--directive-keywords . nil ))
+  )
+
+(defun jenkinsfile-file-compeletion-at-point ()
+  "completion for file step"
+  (interactive)
+  (let* (
+         (bds (bounds-of-thing-at-point 'symbol))
+         (start (car bds))
+         (end (cdr bds)))
+    (list start end jenkinsfile-mode--file-section-keywords . nil ))
+  )
+
 ;;;###autoload
 (define-derived-mode jenkinsfile-mode groovy-mode "Jenkinsfile"
   "Major mode for editing Jenkins declarative pipeline files."
-  (setq font-lock-defaults '(jenkinsfile-mode-font-lock-defaults)))
+  (setq font-lock-defaults '(jenkinsfile-mode-font-lock-defaults))
+  (add-hook 'completion-at-point-functions 'jenkinsfile-file-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-directive-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-option-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-pipeline-step-compeletion-at-point nil 'local)
+  (add-hook 'completion-at-point-functions 'jenkinsfile-core-step-compeletion-at-point nil 'local)
```

Comment:
Disclaimer: I'm not too familiar with how completion-at-point works. Do these `add-hook` calls need an equivalent `remove-hook` call somewhere else?